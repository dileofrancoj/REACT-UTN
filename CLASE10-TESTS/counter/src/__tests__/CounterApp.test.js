import React from "react";
import { shallow } from "enzyme"; // acceder al dom de un componente
import CounterApp from "./../CounterApp";
describe("Pruebas de CounterApp con valor 10 por default", () => {
  // pruebas unitarias
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  // El valor por default
  test("Debe cargar el CounterApp", () => {
    // observer -> publish suscriber -> snapshot
    expect(wrapper).toMatchSnapshot(); // true -> false
  });

  // getElementByClassName, TAG, id
  test("Debe mostrar el valor de 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const text = wrapper.find("h2").text().trim();
    expect(text).toBe("100");
  });

  test("Debe incrementar en 1", () => {
    const expectedOutput = "11";
    // queryselect // getElementByTagName
    wrapper.find("button").at(0).simulate("click");
    const text = wrapper.find("h2").text().trim();
    expect(text).toBe(expectedOutput);
  });
  test("Debe decrementar en 1", () => {
    const expectedOutput = "9";
    // queryselect // getElementByTagName
    wrapper.find("button").at(2).simulate("click");
    const text = wrapper.find("h2").text().trim();
    expect(text).toBe(expectedOutput);
  });

  test("Debe volver a 10", () => {
    wrapper.find("button").at(1).simulate("click");
    const text = wrapper.find("h2").text().trim();
    expect(text).toBe("10");
  });
  // React-testing-library (XSS + SQLi) '
});
