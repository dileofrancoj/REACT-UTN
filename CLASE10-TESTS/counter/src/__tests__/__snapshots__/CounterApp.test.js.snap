// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de CounterApp Debe cargar el CounterApp 1`] = `ShallowWrapper {}`;

exports[`Pruebas de CounterApp con valor 10 por default Debe cargar el CounterApp 1`] = `ShallowWrapper {}`;
